### DISCUSSION TITLE
Imports of packages in External repos

### DISCUSSION THREAD

**Discussion by SindreBrurberg** [cue-lang/cue#2705] [Q&A] [answered]
Hi, I will start by apologising for my "noob" question, but I think that I am slightly confused and in need of some help. After reading the docs on packages (https://cuelang.org/docs/concepts/packages/) I got the understanding that I could define a set of definitions that I later would be able to import into other projects. I am capable of creating my module, and everything works exactly as expected within that module. My issue comes when I want to use that module in another git repo. What is the intended way to do that? Mono repo? Git submodules? Or am I going down the complete wrong path?

Our usecase for this is a custom created deployment solution where different teams / customers should have access to their own configurations, but not be able to edit the base definitions used by the deployment software.

**Comment by mvdan** [comment, marked as answer]
Being able to publish and download modules is something we've been actively working on :) We will have an experimental version to share soon. You can subscribe to https://github.com/cue-lang/cue/issues/2032 for updates, and see https://github.com/cue-lang/cue/discussions/2330 for our latest proposed design.

**Reply by SindreBrurberg** [reply]
Just noting this here as I think I have found a acceptable solution for the mean time. This is a Go specific solution, however it allows me to continue with the "demo" / experiment we are doing for now.
By using the following package you can both embed a schema and load defaults. 
[github.com/cue-exp/cueconfig](https://github.com/cue-exp/cueconfig)
In my case I would probably load the defaults from a external service. And embed the schema. The current interface seems a bit too constrictive right now, so if my experiments work out, I'll finish the fork adding possibility to embed a full FS for schema. I am not yet 100% sure if this all will work how I am envisioning it. However if it does, then cue will be the defacto config language going forward.

And lastly, thank you for your answer. I have subscribed to those two and will keep an eye our if some other solutions pop up.
